src/overlays/actors/ovl_En_Wf/z_en_wf.c

```diff
@@ -137,81 +144,1338 @@ static ColliderCylinderInit D_80B37A5C = {
+                    }
+                }
+            }
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B34380.s")
+void EnWf_Update(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnWf* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B34428.s")
+    EnWf_UpdateDamage(this, globalCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B3455C.s")
+    if (this->actor.colChkInfo.damageEffect != ENWF_DMGEFF_ICE_MAGIC) {
+        Actor_MoveForward(&this->actor);
+        Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 32.0f, 30.0f, 60.0f, 29);
+        this->actionFunc(this, globalCtx);
+        EnWf_FaceTowardsPlayer(this, globalCtx);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B345E4.s")
+    if (this->actor.bgCheckFlags & (1 | 2)) {
+        func_800359B8(&this->actor, this->actor.shape.rot.y, &this->actor.shape.rot);
+    } else {
+        Math_SmoothStepToS(&this->actor.shape.rot.x, 0, 1, 1000, 0);
+        Math_SmoothStepToS(&this->actor.shape.rot.z, 0, 1, 1000, 0);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B347FC.s")
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->colliderSphere.base);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B3487C.s")
+    if (this->action >= WOLFOS_ACTION_WAIT) {
+        if ((this->actor.colorFilterTimer == 0) || !(this->actor.colorFilterParams & 0x4000)) {
+            Collider_UpdateCylinder(&this->actor, &this->colliderCylinder1);
+            CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->colliderCylinder2.base);
+            CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->colliderCylinder1.base);
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B34CFC.s")
+    if (this->action == WOLFOS_ACTION_REACT_TO_PLAYER) {
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->colliderSphere.base);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B34D48.s")
+    if (this->slashStatus > 0) {
+        if (!(this->colliderSphere.base.atFlags & AT_BOUNCED)) {
+            CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->colliderSphere.base);
+        } else {
+            EnWf_SetupSecondSlash(this);
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B34F28.s")
+    this->actor.focus.pos = this->actor.world.pos;
+    this->actor.focus.pos.y += 25.0f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B35024.s")
+    if (this->eyeIndex == 0) {
+        if ((Rand_ZeroOne() < 0.2f) && ((globalCtx->gameplayFrames % 4) == 0) && (this->actor.colorFilterTimer == 0)) {
+            this->eyeIndex++;
+        }
+    } else {
+        this->eyeIndex = (this->eyeIndex + 1) & 3;
```

Comment:
```suggestion
        this->eyeIndex = (this->eyeIndex + 1) % 4;
```
my favourite lottery

Reply:
I've never had this sort of thing work unless it's 2^n - 1 and being compared to 0. And that's the case here too.